### PR TITLE
Remove excluded archs & xcode 12.5 patch

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

This removes the excluded archs now that upgrades to MacOS 13.1 seem to be able to run an arm64 simulator.
We should be careful merging this because it will make builds fails if you are even on 13.01!

Requires an `npm i`

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Locally, failing on 12.6, working on 13.1

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/ios/Podfile [hunk 1]
```diff
@@ -49,7 +49,6 @@ target 'AudiusReactNative' do
       # necessary for Mac Catalyst builds
       :mac_catalyst_enabled => false
     )
-    __apply_Xcode_12_5_M1_post_install_workaround(installer)
```
Comment (rickyrombo) [inline packages/mobile/ios/Podfile:52]: https://react-native-community.github.io/upgrade-helper/?from=0.66.5&to=0.70.6 it seems like it only applies to older Xcode, do we really want to remove?

File: packages/mobile/ios/AudiusReactNative.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -453,7 +453,6 @@
 				CODE_SIGN_ENTITLEMENTS = "AudiusReactNative/Audius Music.entitlements";
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_TEAM = LRFCG93S85;
-				EXCLUDED_ARCHS = "";
```
Comment (rickyrombo) [inline packages/mobile/ios/AudiusReactNative.xcodeproj/project.pbxproj:456]: Are we sure we want to remove these? @dylanjeffers already removed the arm64 exclusion

**Review by rickyrombo — COMMENTED** [review]
Not sure if we really need this

File: packages/mobile/ios/Podfile (hunk 1 above)
Comment (raymondjacobson) [reply packages/mobile/ios/Podfile:52]: won't pod install exclude arm64 after you run it?

Comment (rickyrombo) [reply packages/mobile/ios/Podfile:52]: I don't think so - that was a different couple lines that did it manually

**Comment by dylanjeffers** [comment]
based on learnings, it looks like we have to exclude arm64 arch from simulator since some 3rd party pods only provide an x86 variant for simulator

**Pull request closed without merging**